test/cppsuite/tests/bounded_cursor_perf.cpp

```diff
@@ -0,0 +1,161 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/workload/random_generator.h"
+#include <time.h>
+#include "test_harness/core/op_tracker.cpp"
+
+namespace test_harness {
+
+/*
+ * Class that defines operations that do nothing as an example. This shows how database operations
+ * can be overridden and customized.
+ */
+class bounded_cursor_perf : public test {
+
+    enum set_bounds { SET_LOWER, SET_UPPER, SET_ALL };
+    std::string test_name;
+
+    public:
+    bounded_cursor_perf(const test_args &args) : test(args) {
+        this->test_name = args.test_name;
```

Comment:
Instead of saving the test name in the constructor what you can do instead is: 
```
diff --git a/test/cppsuite/test_harness/test.h b/test/cppsuite/test_harness/test.h
index f5df7902f..cc6ef5b4f 100644
--- a/test/cppsuite/test_harness/test.h
+++ b/test/cppsuite/test_harness/test.h
@@ -83,12 +83,12 @@ class test : public database_operation {
     thread_manager *get_thread_manager();

     protected:
+    const test_args &_args;
     configuration *_config;
     timestamp_manager *_timestamp_manager = nullptr;
     workload_tracking *_workload_tracking = nullptr;

     private:
-    const test_args &_args;
     std::vector<component *> _components;
     checkpoint_manager *_checkpoint_manager = nullptr;
     runtime_monitor *_runtime_monitor = nullptr;
```
Then in your class where you need it you just do: `test::_args.test_name`.